app/controllers/components/course/controller_component_host.rb

```diff
@@ -52,6 +53,25 @@ def gamified?
       def can_be_disabled?
         true
       end
+
+      # Returns a model which the current component can use to interface with its persisted
+      # settings.
+      #
+      # Example:
+      # If the component Course::FoobarComponent has settings, define a class
```

Comment:
I think it's better to leave it near where the code is defined, otherwise it is very easy to forget to change the documentation when the code changes. Have added documentation on creating components at the top level instead. Could you see if you are okay with it?